### ISSUE TITLE
Double free in libvapoursynth.so crashing vspipe

### ISSUE THREAD

**Issue by putnam** [vapoursynth/vapoursynth#873]
I'm currently using R58 and latest havsfunc (and its APIV4 compatible deps) on Linux. There is a double free occurring in libvapoursynth.so which crashes vspipe:

Script:
```
import sys
from vapoursynth import core
sys.path.append('/opt/vapoursynth/share/vsscripts')
import havsfunc as haf

infile = "/tmp/test.mkv"
video = core.ffms2.Source(source=infile)
video = haf.QTGMC(video, Preset='Medium', TFF=True)
video.set_output()
```

When vspipe is called:
```
$ vspipe myscript.vpy -
double free or corruption (!prev)
```

If I comment out the QTGMC call, vspipe behaves as expected.

Backtrace:
```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:49
#1  0x00007ffff7bae546 in __GI_abort () at abort.c:79
#2  0x00007ffff7c05eb8 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7d23a78 "%s\n")
    at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007ffff7c0d91a in malloc_printerr (str=str@entry=0x7ffff7d25fa0 "double free or corruption (!prev)") at malloc.c:5628
#4  0x00007ffff7c0f1ac in _int_free (av=0x7fffb0000020, p=0x7fffb009c010, have_lock=<optimized out>) at malloc.c:4550
#5  0x00007ffff7c129b4 in __GI___libc_free (mem=<optimized out>) at malloc.c:3309
#6  0x00007ffff6c67ca8 in vs_generic_1d_conv_x_byte_avx2 () from /opt/vapoursynth/lib/libvapoursynth.so
#7  0x00007ffff6b61c29 in VSFrame const* genericGetframe<(GenericOperations)7>(int, int, void*, void**, VSFrameContext*, VSCore*, VSAPI const*) () from /opt/vapoursynth/lib/libvapoursynth.so
#8  0x00007ffff6bcc91e in VSNode::getFrameInternal(int, int, VSFrameContext*) () from /opt/vapoursynth/lib/libvapoursynth.so
#9  0x00007ffff6bdabde in VSThreadPool::runTasks(std::atomic<bool>&) () from /opt/vapoursynth/lib/libvapoursynth.so
#10 0x00007ffff7e599d3 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#11 0x00007ffff744bd80 in start_thread (arg=0x7fffcb7fe640) at pthread_create.c:481
#12 0x00007ffff7c8676f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by putnam** [comment]
Potentially related: #868

**Comment by putnam** [comment]
After compiling head, the problem went away. Ran a bisect and it was fixed by the same patch that fixed #868 (0373efd8bab0a6a60f447e33946aef4129d634bd)

Could be worth a glance before I close this just in case that is a coincidental fix.

**Comment by myrsloik** [comment]
Definitely memory corruption and fixed
